modules-js/react-fleet/src/form-elements/UploadPhoto.tsx

```diff
@@ -0,0 +1,285 @@
+            {({ getRootProps, getInputProps, isDragActive }) => (
+              <div
+                {...getRootProps()}
+                style={{ height: this.state.previewHeight }}
+              >
+                <input {...getInputProps()} />
+
+                <div
+                  ref={this.previewRef}
+                  className={`${PREVIEW_CONTAINER_STYLING} ${
+                    isDragActive ? DRAG_RING_STYLING : ''
+                  }`}
+                >
+                  {file ? (
+                    <img
+                      src={file.preview}
+                      alt=""
+                      className={PREVIEW_IMAGE_STYLING}
+                      style={{ maxHeight: this.state.previewHeight }}
+                    />
+                  ) : (
+                    <>
+                      {this.props.backgroundElement ||
+                        defaultInitialAppearance()}
+                    </>
+                  )}
+                </div>
+              </div>
+            )}
+          </Dropzone>
+
+          <div className="br br-t200" style={{ position: 'relative' }}>
+            <button
+              className={`btn btn--w btn--b ${isUploading ? 'btn--r-hov' : ''}`}
+              style={{ width: '100%' }}
+              type="button"
+              onClick={this.handleButtonClick}
+            >
+              {buttonTitle}
+            </button>
+
+            {isUploading && (
+              <div
+                className={PROGRESS_STYLING}
+                style={{ width: `${uploadProgress}%` }}
+              />
+            )}
+          </div>
+        </div>
+
+        {hasError && (
+          <div className={`t--info p-a300 ${ERROR_MESSAGE_STYLING}`}>
+            Error: {errorMessage}
+          </div>
+        )}
+      </>
+    );
+  }
+}
+
+/**
+ * Defaults to a camera icon.
+ */
+function defaultInitialAppearance(): JSX.Element {
+  return (
+    <div className={DEFAULT_IMAGE_STYLING}>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        viewBox="0 22 144 100"
+        aria-hidden="true"
+        focusable="false"
+      >
+        <polygon points="93.58 51.5 87.75 41.5 59.55 41.5 54.35 51.5 27.18 51.5 27.18 102.5 116.82 102.5 116.82 51.5 93.58 51.5" />
+        <rect x="31.48" y="44.49" width="15.23" height="7.01" />
+        <circle cx="74" cy="74.38" r="22.88" />
+        <circle cx="74" cy="74.38" r="17.47" />
+        <circle cx="38.63" cy="60.15" r="3.24" />
+        <line x1="96.12" y1="67.86" x2="116.82" y2="67.86" />
+        <line x1="27.18" y1="67.86" x2="51.88" y2="67.86" />
+      </svg>
+    </div>
+  );
+}
+
+const DEFAULT_IMAGE_STYLING = css({
+  backgroundColor: GRAY_300,
+  color: WHITE,
+  svg: {
+    strokeWidth: 3,
+    strokeLinecap: 'round',
+    fill: 'none',
+    stroke: 'currentColor',
+  },
+});
+
+const PREVIEW_CONTAINER_STYLING = css({
+  position: 'relative',
+  cursor: 'pointer',
+  overflow: 'hidden',
+  backgroundColor: 'currentColor',
```

Comment:
HOW HAVE I NEVER HEARD OF `currentColor` BEFORE??? Love it.

Reply:
It is the best thing ever. The **best**